packages/react/src/components/table/table-header.tsx

```diff
@@ -0,0 +1,142 @@
+import { ReactElement } from 'react';
+import styled, { css } from 'styled-components';
+import {
+    Header,
+    HeaderGroup,
+    ColumnDef as OriginalColumnDef,
+    Column,
+    flexRender,
+} from '@tanstack/react-table';
+import { SortButtonIcon, SortState } from './sort-button-icon';
+
+type TextAlignOptions = 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
+
+type CustomColumnDef<TData extends object, TValue> = OriginalColumnDef<TData, TValue> & {
+    className?: string;
+    textAlign?: TextAlignOptions; // Define this type if not already defined
+    sticky?: boolean;
+    stickyHeader?: boolean;
+    position?: number;
+    sortable?: boolean;
+    iconAlign?: 'left' | 'right';
+    defaultSort?: 'asc' | 'desc';
+};
+
+interface CustomHeader<TData extends object, TValue> extends Header<TData, TValue> {
+  column: Column<TData, TValue> & {
+      columnDef: CustomColumnDef<TData, TValue>;
+  };
+  sortable?: boolean;
+  headerColSpan?: number;
+  iconAlign?: 'left' | 'right';
+}
+
+const SortButton = styled.button<{ textAlign: string }>`
+    align-items: center;
+    cursor: pointer;
+    display: flex;
+    margin-left: var(--spacing-1x);
+    ${({ textAlign }) => textAlign && `text-align: ${textAlign};`}
+
+    &:focus {
+        outline: none;
+    }
+`;
+
+const StyledHeader = styled.th<{ sticky: boolean, position: number }>`
```

Comment:
Position is a bad term for this as it mean something else in CSS. Also position is not enough clear, position where? top/left/right?

Rename to `startOffset` (I took this from the description of what `header.getStart()` is returning)